components/payments/settings/payments-webhooks-table.tsx

```diff
@@ -0,0 +1,311 @@
+import {
+  PaymentsWebhooksType,
+  isValidWebhookUrl,
+} from "@3rdweb-sdk/react/hooks/usePayments";
+import {
+  ButtonGroup,
+  FormLabel,
+  FormControl,
+  Icon,
+  IconButton,
+  Input,
+  Modal,
+  Tr,
+  Td,
+  Th,
+  Thead,
+  Tbody,
+  Table,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  Flex,
+  useDisclosure,
+  Stack,
+  Center,
+  Spinner,
+} from "@chakra-ui/react";
+import {
+  ForwardedRef,
+  forwardRef,
+  useRef,
+  useMemo,
+  useEffect,
+  useState,
+} from "react";
+import pluralize from "pluralize";
+import { format } from "date-fns";
+import { Button, Text } from "tw-components";
+import { BiPencil } from "react-icons/bi";
+import { FaCheck } from "react-icons/fa";
+import { FiTrash2 } from "react-icons/fi";
+import { IoMdAdd } from "@react-icons/all-files/io/IoMdAdd";
+import { FaXmark } from "react-icons/fa6";
+import { TableContainer } from "tw-components";
+
+export interface PaymentsWebhooksTableProps {
+  webhooks: PaymentsWebhooksType[];
+  onCreate: (url: string) => void;
+  onUpdate: (webhook: PaymentsWebhooksType, newUrl: string) => void;
+  onDelete: (webhook: PaymentsWebhooksType) => void;
+  isLoading: boolean;
+  isFetched: boolean;
+}
+
+interface UrlInputFieldProps {
+  webhookUrl: string;
+  onChange: (value: string) => void;
+}
+
+const UrlInputField = forwardRef<HTMLInputElement, UrlInputFieldProps>(
+  ({ webhookUrl, onChange }, ref) => {
+    const isValid = isValidWebhookUrl(webhookUrl);
+    return (
+      <FormControl isInvalid={!isValid && !!webhookUrl}>
+        <Input
+          ref={ref}
+          type="url"
+          placeholder="Webhook URL"
+          value={webhookUrl}
+          onChange={(e) => onChange(e.target.value)}
+          autoFocus
+        />
+      </FormControl>
+    );
+  },
+);
+
+interface EditTableRowProps {
+  webhook: PaymentsWebhooksType;
+  onEdit: (webhook: PaymentsWebhooksType, newUrl: string) => void;
+  onDelete: (webhook: PaymentsWebhooksType) => void;
+}
+
+const EditTableRow = ({ webhook, onEdit, onDelete }: EditTableRowProps) => {
+  const [editUrl, setEditUrl] = useState(webhook.url ?? "");
```

Comment:
I still wonder if we really need state here for this but we can look over it tomorrow, we might do but we might find a cleaner way

Reply:
I changed this from a ref because renders were clearing all the input fields